locator/tablets.hh

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright (C) 2023-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#pragma once
+
+#include "dht/token.hh"
+#include "utils/small_vector.hh"
+#include "locator/host_id.hh"
+#include "dht/i_partitioner_fwd.hh"
+#include "schema/schema_fwd.hh"
+#include "utils/chunked_vector.hh"
+#include "utils/hash.hh"
+
+#include <seastar/core/reactor.hh>
+#include <seastar/util/log.hh>
+
+#include <vector>
+
+namespace locator {
+
+extern seastar::logger tablet_logger;
+
+using token = dht::token;
+
+// Identifies tablet within the scope of a single tablet_map,
+// which has a scope of (table_id, token metadata version).
+// Different tablets of different tables can have the same tablet_id.
+// Different tablets in subsequent token metadata version can have the same tablet_id.
+// When splitting a tablet, one of the new tablets (in the new token metadata version)
+// will have the same tablet_id as the old one.
+using tablet_id = size_t;
+
+struct tablet_replica {
+    host_id host;
+    shard_id shard;
+
+    bool operator==(const tablet_replica&) const = default;
+};
+
+std::ostream& operator<<(std::ostream&, const tablet_replica&);
+
+using tablet_replica_set = utils::small_vector<tablet_replica, 3>;
```

Comment:
I'd like this to be std::span pointing into a larger space, to avoid lots of small allocations. This minor optimization isn't needed now, and the code should be well isolated.